### DISCUSSION TITLE
Jekyll Markdown abbr

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#10242] [Q&A] [answered]
in Jekyll, you can use syntax like this at the end of a markdown post:

```
*[CSS]: Cascading Style Sheets
```

and each time you use `CSS` it compiles into:

```html
<abbr title="Cascading Style Sheets">CSS</abbr>
```

is there something like this in Docusaurus? Or how to get similar behavior?

**Comment by OzakIOne** [comment]
I'm not aware of anything like this, maybe because it is not in the official spec of markdown (I don't know)

However, natively you have footnotes if that might work for you

```
Here is a simple footnote[^1]. With some additional text after it.

[^1]: My reference.
```

**Reply by slorber** [reply]
Yes, footnotes should be the equivalent.

Eventually, we'll improve the UX around accessing the footnote content. 
Example: https://frontendmasters.com/blog/footnotes-progressively-enhanced-to-popovers/

**Reply by jcubic** [reply]
Footnotes are useful, but not something that I would like to use in case of abbreviation. The semantic tag for this is an `<abbr>`.

Is there a way to extend Markdown like in Jekyll or Eleventy? Or the only option is MDX?

**Reply by slorber** [reply, marked as answer]
We don't provide things like this out of the box, but you can implement many things like this with Remark plugins

You might find interesting:
- https://gist.github.com/wooorm/e5347ee97569bf6f7a079357c419cf7e
- https://github.com/renatonagliati/remark-auto-glossary

If the behavior is not exactly what you want, you can still get inspired by the code, fork it, and tweak it to your needs.

In @wooorm example, he recommends:

> Re extending markdown? You most likely shouldn't: [github.com/micromark/micromark#extending-markdown](https://github.com/micromark/micromark#extending-markdown)

> I recommend going with directives if you can!

Parsing `*[CSS]: Cascading Style Sheets` is more complicated to implement and not std Markdown

If you are ok with an alternative syntax for abbreviation titles, we already parse remark directives for you, so technically you could implement a remark plugin that supports any of those syntaxes:

```md
::abbr[CSS]{title="Cascading Style Sheets"}

::abbr[Cascading Style Sheets]{source="CSS"}

::abbr{source="CSS" title="Cascading Style Sheets"}

::CSS[Cascading Style Sheets]


:::abbr[CSS]

Cascading Style Sheets

:::
```

Those syntaxes are "pre-parsed" for you thanks to remark-directive (that we already use). See the mdx playground:

![CleanShot 2024-06-27 at 14 03 44](https://github.com/facebook/docusaurus/assets/749374/850d1abb-7848-4729-85bf-262e7a633edb)



You are then free to pick the syntax you find the most convenient and implement a remark plugin similar to the pseudo example provided, that creates a registry to abbrs, and then find/replace text that matches any registered abbreviation.

Note that there are other possibilities than using directives, for example you could also support something like:

```jsx
export abbr = {
  CSS: "Cascading Style Sheets",
  HTML: "HyperText Markup Language"
}
```

And you might even improve the UX by using an interactive progressively enhanced `<Abbr>` React component instead of just `<abbr>`


---

Funnily enough, the official doc of remark-directive use abbr as examples:
- https://github.com/remarkjs/remark-directive#use
- https://github.com/micromark/micromark-extension-directive#use

(although you probably don't want to write `A lovely language know as :abbr[HTML]{title="HyperText Markup Language"}` this can give you inspiration.
